http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcBaseTest.java

```diff
@@ -394,7 +428,31 @@ private void performAuthentication(InputStream oidcConfig, String username, Stri
             }
 
             if (loginToKeycloak) {
-                client.setDispatcher(createAppResponse(mechanism, expectedDispatcherStatusCode, expectedLocation, clientPageText));
+                // change the sessionRandomValue value so that the compare to nonce will fail.
+                if (changeSessionId) {
+                    HttpServerCookie stateCookie = null;
+                    String sessionIdStr = null;
+                    List<HttpServerCookie> tmpCookies = response.getCookies();
+                    for (int i = 0; i < tmpCookies.size(); i++) {
+                        if (tmpCookies.get(i).getName().equals(oidcClientConfiguration.getStateCookieName())) {
+                            stateCookie = tmpCookies.get(i);
+                        } else if (tmpCookies.get(i).getName().equals(SESSION_RANDOM_VALUE)) {
+                            sessionIdStr = "; " + tmpCookies.get(i).getName() + "=99" + tmpCookies.get(i).getName();
```

Comment:
I'm wondering if the changes in this method are causing the concatenation issue you mentioned above.

Reply:
No this is not the issue